feat(recurringjob): add RecurringJobMaxRetention setting (#7390)
longhorn/longhorn: chart/README.md

| defaultSettings.orphanAutoDeletion | This setting allows Longhorn to delete the orphan resource and its corresponding orphaned data automatically like stale replicas. Orphan resources on down or unknown nodes will not be cleaned up automatically. |
| defaultSettings.priorityClass | priorityClass for Longhorn system-managed components This setting can help prevent Longhorn components from being evicted under Node Pressure. Notice that this will be applied to Longhorn user-deployed components by default if there are no priority class values set yet, such as `longhornManager.priorityClass`. |
| defaultSettings.recurringFailedJobsHistoryLimit | This setting specifies how many failed backup or snapshot job histories should be retained. History will not be retained if the value is 0. |

> | defaultSettings.recurringJobMaxRetention | This setting specifies how many snapshots or backups should be retained |

Reviewer: ```suggestion
| defaultSettings.recurringJobMaxRetention | Maximum number of snapshots or backups to be retained. |
```

Author: Modified. Thanks
